File: src/main/java/no/stunor/origo/batch/data/RegionRepository.java
package no.stunor.origo.batch.data;


import org.springframework.stereotype.Repository;

import com.google.cloud.spring.data.firestore.FirestoreReactiveRepository;

import no.stunor.origo.batch.model.Region;
import reactor.core.publisher.Mono;

@Repository
public interface RegionRepository extends FirestoreReactiveRepository<Region> {
    Mono<Region> findByRegionIdAndEventorId(String regionId, String eventorId);
    //void deleteWithLastUpdatedBefore(Timestamp timestamp);
}

File: src/main/kotlin/no/stunor/origo/batch/data/RegionRepository.kt
package no.stunor.origo.batch.data

import no.stunor.origo.batch.model.Region
import org.springframework.data.repository.CrudRepository
import org.springframework.stereotype.Repository
import java.util.UUID

@Repository
interface RegionRepository : CrudRepository<Region, UUID> {
    fun findAllByEventorId(eventorId: String): List<Region>
}